prototype2/multiblade/mbcaen/DataParser.cpp

```diff
@@ -0,0 +1,48 @@
+/** Copyright (C) 2017-2018 European Spallation Source */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Class to parse detector readout for Multi-Blade using Jadaq UDP
+/// Data format is described in
+/// https://github.com/ess-dmsc/jadaq/blob/devel/DataFormat.hpp
+///
+//===----------------------------------------------------------------------===//
+
+#include <common/Trace.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <mbcaen/DataParser.h>
+#include <memory>
+
+#undef TRC_LEVEL
+#define TRC_LEVEL TRC_L_DEB
+
+int DataParser::parse(const char *buffer, unsigned int size) {
+
+  MBHeader = nullptr;
+  MBData = nullptr;
+  memset(&Stats, 0, sizeof(struct Stats));
+
+  auto headerlen = sizeof(struct Header);
+  if (size < headerlen) {
+    XTRACE(DATA, WAR, "Invalid data size: received %d, min. expected: %lu", size, headerlen);
+    Stats.error_bytes += size;
+    return -error::ESIZE;
+  }
+
+  MBHeader = (struct Header *)buffer;
```

Comment:
I would suggest using `reinterpret_cast` here instead of C-style cast

Reply:
Why? I like the clarity and hate how reinterpret cast makes the code less 
readable.

Reply:
I think it actually makes it more readable, and more easily searcheable. There are things out the on the intewebs that will tell you more about the differences, but mainly that and it is safer, fo example:
https://stackoverflow.com/questions/7831696/reinterpret-cast-vs-c-style-cast